src/bufr/DataContainer.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * (C) Copyright 2020 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+
+#include <map>
+#include <string>
+#include "Eigen/Dense"
+
+#include "DataContainer.h"
+
+
+namespace Ingester
+{
+    void DataContainer::add(const std::string& fieldName, const std::shared_ptr<DataObject> data)
+    {
+        if (hasKey(fieldName))
+        {
+            std::cout << "ERROR: Field called " << fieldName << " already exists." << std::endl;
+            abort();
```

Comment:
Throw an eckit::Exception. ```abort``` is a C function that does not process ```atexit``` hooks. We use eckit loggers that we want to flush to the console even if an error has occurred.